book/external_language_integration/julia/nimjl_arrays.nim

```diff
@@ -0,0 +1,364 @@
+
+Case 1 : Using Tensor argument directly (no copy):
+"""
+
+nbCodeInBlock:
+  var res = Julia.transpose(orderedTensor).toJlArray(int)
+  echo res
+  echo orderedTensor.transpose()
+
+nbText:"""This is expanded to:
+"""
+
+nbCodeInBlock:
+  # When passing localTensor, a ``JlArray`` is created using ``jlFromBuffer``.
+  # Since the Tensor is row major and the Array col major, the order of the values is not conserved
+  var res = Julia.transpose(toJlVal(jlArrayFromBuffer(orderedTensor))).toJlArray(int)
+  echo res
+  echo orderedTensor.transpose()
+
+nbText:"""Therefore, no copy is made : the Julia Array points to the Tensor's buffer.
+
+The indexed values between ``Julia.transpose(...)`` and ``orderedTensor.transpose()`` **are different** because they are indexed differently : Julia Arrays are indexed in column major while this Arraymancer Tensor is in column major.
+
+Case 2 : Copying the Tensor into an Array and using the Array:
+"""
+nbCodeInBlock:
+  var tensorCopied = toJlArray(orderedTensor)
+  # Tensor is copier to Array in ColMajor order
+  var res = Julia.transpose(tensorCopied).toJlArray(int)
+  echo res
+  echo orderedTensor.transpose()
+
+nbText:"""On the other hand, on this case because the Array has been created from **a copy**, the indexed value have been copied into ``JlArray`` in column major order.
+
+As a consequence, the indexed value of ``Julia.transpose()`` and ``orderedTensor.transpose()`` **are identical**.
+
+Note that you can use ``swapMemoryOrder`` on an existing ``JlArray[T]`` to obtain a copy of the Array but permuted.
+"""
+
+nbCodeInBlock:
+  var tensorView = jlArrayFromBuffer(orderedTensor)
+  var tensorCopied = toJlArray(orderedTensor)
+  echo tensorView
+  echo tensorCopied
+
+nbText: """The array are actually different from Julia's point of view: ``tensorView`` is row major values (the Tensor buffer) indexed as column major while ``tensorCopied`` is col major values indexed as col major.
+
+In Nim, the utility proc ``swapMemoryOrder()`` will change **return a copy** with a swapped memory order (col major -> row major & vice-versa) to handle such cases more easily.
+
+## Broadcasting
+
+One of main appeal of Arrays in Julia, is the ability to broadcast function of a single element.
+In Nimjl this is done using the ``jlBroadcast``.
+"""
+
+nbCodeInBlock:
+  var localArray = @[
+    @[4, 4, 4, 4],
+    @[4, 4, 4, 4],
+    @[4, 4, 4, 4]
+  ].toJlArray()
+  echo localArray
+  var sqrtLocalArray = jlBroadcast(sqrt, localArray).toJlArray(float) # sqrt of int is a float
+  echo sqrtLocalArray
+
+nbText: """This is the equivalent in Julia of calling ``sqrt.(localArray)``.
+
+For convenience, the usual broadcasted operators have also been defined:
+"""
+
+nbCodeInBlock:
+  var localArray = @[
+    @[4, 4, 4, 4],
+    @[4, 4, 4, 4],
+    @[4, 4, 4, 4]
+  ].toJlArray()
+  echo localArray
+  var res = (localArray +. localArray)*.2 -. (localArray/.2)
+  echo res
+
+nbText: """## Final word ?
```

Comment:
Out of curiosity, is it the norm to include a space before punctuation (ex: `Final word ?` instead of `Final word?`) in your native language? :)

Reply:
Eh, I think so but to be perfectly honest, I'm not sure. Maybe it's just a bad habit I picked up.